File: frontend/src/types/question.ts
export type QuestionType = 'single' | 'multi' | 'judge' | 'fill' | 'essay';

export interface Option {
  label: string;
  content: string;
  is_correct?: boolean;
}

export interface ParsedQuestion {
  temp_id: string;
  type: QuestionType;
  content: string;
  options?: Option[];
  answer: string;
  score: number;
  explanation?: string;
}

export interface ParseResult {
  filename: string;
  file_path: string;
  file_size: number;
  total_count: number;
  questions: ParsedQuestion[];
  type_summary: Record<QuestionType, number>;
}

export interface Question {
  id: number;
  type: QuestionType;
  content: string;
  score: number;
  created_at: string;
}

export interface QuestionDetail extends Question {
  options?: Option[];
  answer: string;
  explanation?: string;
}

export interface QuestionListResponse {
  items: Question[];
  total: number;
  page: number;
  page_size: number;
}

export interface SelectedQuestion {
  id: number;
  type: QuestionType;
  content: string;
  score: number;
}

File: backend/app/schemas/question.py
"""题目入库请求/响应模型"""

from pydantic import BaseModel, Field


class OptionSaveItem(BaseModel):
    """选项入库项"""
    label: str
    content: str


class QuestionSaveItem(BaseModel):
    """题目入库项"""
    temp_id: str = ""
    type: str = Field(..., description="single | multi | judge | fill | essay")
    content: str
    options: list[OptionSaveItem] = []
    answer: str = ""
    score: float = 0.0
    explanation: str = ""


class QuestionSaveRequest(BaseModel):
    """确认入库请求"""
    filename: str = Field(..., description="原始文件名")
    questions: list[QuestionSaveItem] = Field(..., description="解析后的题目列表")


class QuestionResponse(BaseModel):
    """题目响应"""
    id: int
    type: str
    content: str
    answer: str
    score: float
    explanation: str

    model_config = {"from_attributes": True}


class QuestionSaveResponse(BaseModel):
    """入库响应"""
    exam_id: int
    questions: list[QuestionResponse]


class QuestionListItem(BaseModel):
    """题目列表项"""
    id: int
    type: str
    content: str
    score: float

    model_config = {"from_attributes": True}


class QuestionDetail(BaseModel):
    """题目详情（含选项）"""
    id: int
    type: str
    content: str
    options: list[OptionSaveItem] = []
    answer: str
    score: float
    explanation: str = ""

    model_config = {"from_attributes": True}


class QuestionListResponse(BaseModel):
    """题目列表分页响应"""
    items: list[QuestionListItem]
    total: int
    page: int
    page_size: int
